### ISSUE TITLE
[Bug] open union as one response returns multiple responses

### ISSUE THREAD

**Issue by pshao25** [Azure/typespec-azure#509]
**Describe the bug**
See this case: [Playground Link](https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwpAc2VydmljZSh7CiAgdGl0bGU6ICJXaWRnZXQgU8YbIiwKfSkKbmFtZXNwYWNlIERlbW%2FHG8RWbmlvbiBEYXlzT2ZXZWVrRXh0ZW5zaWJsZUVudW0gxFcgIHN0cmluZyzFDE1vbmRheTogIsYJIsYWVHVlc8YXxwrHGFdlZG7IGskMyDRodXLINccLxxpGcmnGGMYJxxZTYXR1csYYyAvIGnXnAJTGCeQA8goKb3AgZ2V0VW5rbm93blZhbHVlKCk6%2BQDqOw%3D%3D&e=%40typespec%2Fopenapi3&options=%7B%7D)
```
union DaysOfWeekExtensibleEnum {
    string,
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
}

op getUnknownValue(): DaysOfWeekExtensibleEnum; 
```
After I upgrade to the latest TCGC (0.41.0-dev.24), it throws error:
![image](https://github.com/Azure/typespec-azure/assets/97225342/3813b865-955b-4fe7-b14f-5b830a1ca470)

The root cause here is, when compiler reads responses from this operation, it gets 8: each one is the value of this union.
![image](https://github.com/Azure/typespec-azure/assets/97225342/7ae7c901-90c2-4901-a187-4994b6618769)

Open question: how should we look at this response? It should be one enum response, or 8 responses with 1 string and 7 constants?

**Comment by tadelesh** [comment]
I doubt if it could be resolved by TCGC since we get multiple http responses for such tsp from http compile api.

**Comment by timotheeguerin** [comment]
yeah it is definitely not a wrong usage, I am fine if you want to add `@body` in cadl-ranch to unblock as this exact case should be quite an edge case.

Fixing in http library might be more tricky on what logic we do to decide to group within the same response.
For example 
```
op test(): "a" | "b" | Error;
```

is this 3 response or 2 ("a"|"b" and Error).
What if this union of 3 variant had a name but "a"| "b" didn't do we then just lose response name.
